resources/tentative/todomvc/src/components/todo-app/todo-app.component.js

```diff
@@ -0,0 +1,164 @@
+import template from "./todo-app.template.js";
+import TodoTopbar from "../todo-topbar/todo-topbar.component.js";
+import TodoList from "../todo-list/todo-list.component.js";
+import TodoBottombar from "../todo-bottombar/todo-bottombar.component.js";
+import { useRouter } from "../../hooks/useRouter.js";
+
+// import { data } from "../../data.js";
+
+class TodoApp extends HTMLElement {
+    constructor() {
+        super();
+
+        // elements
+        this.topbar = undefined;
+        this.list = undefined;
+        this.bottombar = undefined;
+        // state
+        this.connected = false;
+        this._isReady = false;
+        // this._data = [...data];
+        this._data = [];
+
+        this.router = useRouter();
+
+        this.shadow = this.attachShadow({ mode: "open" });
+    }
+
+    get isReady() {
+        return this._isReady;
+    }
+
+    getInstance() {
+        return this;
+    }
+
+    addItem = (e) => {
```

Comment:
Why are these instance variables??

Reply:
are you referring to `get isReady` & `getInstance` ?

I am currently not using `get isReady`, but that could be used to tell the runner when this app is ready to go do stuff. 

The `getInstance` method I added to be able to get a reference to the instance from the runner. 
Reason is that the usage of shadow doms makes it harder to just use `querySelector` from within the test steps

Happy to change that, if there are better options.

Reply:
I'm referring to addItem, toggleItem, etc...

Reply:
those are event handlers: https://github.com/WebKit/Speedometer/blob/a02536f4d1dc8c412ba351825628952c151b4405/resources/tentative/todomvc/src/components/todo-app/todo-app.component.js#L101

Reply:
I get that these are event handlers. Why are they instance variables instead of regular member functions?

Reply:
The reason I am using arrow functions instead of binding the functions in the constructor, is for readability. 
I can always refactor.... but I assume it would be mainly the todo-items where there might be a potential performance boost, since there's only one instance of the other components.

Reply:
@flashdesignory : using arrow functions is fine but using arrow functions which are defined as instance variables seems quite odd. I've never seen code that does that in real life.

Reply:
I can refactor to keep the handlers on the prototypes 👍 . 
I'll make that change and push updates.

Reply:
Or we could define those inside constructors. That's fine too. It's just very unusual for arrow functions to appear as class instance variables.

Reply:
changes done